src/primitive/polyhedron.rs

```diff
@@ -169,6 +175,49 @@ where
     }
 }
 
+/// Iterate over polyhedron faces.
+/// Yields a tuple with the positions of the 3 vertices of each face
+pub struct FaceIterator<'a, S: 'a>
+where
+    S: BaseFloat,
+{
+    polyhedron: &'a ConvexPolyhedron<S>,
+    current: usize,
+}
+
+impl<'a, S: 'a> FaceIterator<'a, S>
+where
+    S: BaseFloat,
+{
+    pub fn new(polyhedron: &'a ConvexPolyhedron<S>) -> Self {
```

Comment:
do we actually need this constructor exposed to the user? and if not, we can get away without the function at all

Reply:
True.